include/deal.II/physics/transformations.h

```diff
@@ -0,0 +1,1177 @@
+      * @f[
+      *  \textrm{det} \mathbf{F} \; \left[ \chi^{-1}\left(\bullet\right)^{\#} \right]_{IJKL}
+      *    := \textrm{det} \mathbf{F} \; F^{-1}_{Ii} F^{-1}_{Jj} \left(\bullet\right)^{\#}_{ijkl} F^{-1}_{Kk} F^{-1}_{Ll}
+      * @f]
+      *
+      * @param[in] h The (spatial) Tensor to be operated on
+      * @param[in] F The deformation gradient tensor $\mathbf{F} \left( \mathbf{X} \right)$
+      * @return      $\textrm{det} \mathbf{F} \; \chi^{-1}\left( \mathbf{h} \right)$
+      */
+      template <int dim, typename Number>
+      Tensor<4,dim,Number>
+      pull_back (const Tensor<4,dim,Number> &h,
+                 const Tensor<2,dim,Number> &F);
+
+      /**
+      * Returns the result of the pull back transformation on a rank-4
+      * contravariant SymmetricTensor, i.e. (in index notation)
+      * @f[
+      *  \textrm{det} \mathbf{F} \; \left[ \chi^{-1}\left(\bullet\right)^{\#} \right]_{IJKL}
+      *    := \textrm{det} \mathbf{F} \; F^{-1}_{Ii} F^{-1}_{Jj} \left(\bullet\right)^{\#}_{ijkl} F^{-1}_{Kk} F^{-1}_{Ll}
+      * @f]
+      *
+      * @param[in] h The (spatial) SymmetricTensor to be operated on
+      * @param[in] F The deformation gradient tensor $\mathbf{F} \left( \mathbf{X} \right)$
+      * @return      $\textrm{det} \mathbf{F} \; \chi^{-1}\left( \mathbf{h} \right)$
+      */
+      template <int dim, typename Number>
+      SymmetricTensor<4,dim,Number>
+      pull_back (const SymmetricTensor<4,dim,Number> &h,
+                 const Tensor<2,dim,Number>          &F);
+
+//@}
+    }
+
+    /**
+     * @name Special operations
+     */
+//@{
+
+    /**
+    * Returns the result of applying Nanson's formula for the transformation of
+    * the material surface area element $d\mathbf{A}$ to the current surfaces
+    * area element $d\mathbf{a}$ under the nonlinear transformation map
+    * $\mathbf{x} = \boldsymbol{\varphi} \left( \mathbf{X} \right)$.
+    *
+    * The returned result is the spatial normal scaled by the ratio of areas
+    * between the reference and spatial surface elements, i.e.
+    * @f[
+    *  \mathbf{n} \frac{da}{dA}
+    *    := \textrm{det} \mathbf{F} \, \mathbf{F}^{-T} \cdot \mathbf{N}
+    *     = \textrm{cof} \mathbf{F} \cdot \mathbf{N} \, .
+    * @f]
+    *
+    * @param[in] N The referential normal unit vector $\mathbf{N}$
+    * @param[in] F The deformation gradient tensor $\mathbf{F} \left( \mathbf{X} \right)$
+    * @return      The scaled spatial normal vector $\mathbf{n} \frac{da}{dA}$
+    */
+    template<int dim, typename Number>
+    Tensor<1,dim,Number>
+    nansons_formula (const Tensor<1,dim,Number> &N,
+                     const Tensor<2,dim,Number> &F);
+
+//@}
+  };
+
+}
+
+
+
+#ifndef DOXYGEN
+// ------------------------- inline functions ------------------------
+
+namespace internal
+{
+  namespace Physics
+  {
+    template <int dim, typename Number>
+    inline
+    Tensor<1,dim,Number>
+    transformation_contraction (const Tensor<1,dim,Number> &V,
```

Comment:
I suspect that you don't want these functions to be exported to user space? In that case, just wrap these functions into an anonymous namespace:
```
namespace internal
{
  namespace Physics
  {
    namespace 
    {
      ...
    }
  }
}
```
This also has the advantage that these functions (which may become instantiated in many object files) don't all end up as exported symbols of the object files and consequently of the shared libraries -- and small object file interfaces keep linking reasonably fast.